tests/components/samsungtv/test_media_player.py

```diff
@@ -56,254 +132,429 @@ class UnhandledResponse(Exception):
-
-    def test_media_next_track(self):
-        """Test for media_next_track."""
-        self.device.send_key = mock.Mock()
-        self.device.media_next_track()
-        self.device.send_key.assert_called_once_with("KEY_FF")
-
-    def test_media_previous_track(self):
-        """Test for media_previous_track."""
-        self.device.send_key = mock.Mock()
-        self.device.media_previous_track()
-        self.device.send_key.assert_called_once_with("KEY_REWIND")
-
-    def test_turn_on(self):
-        """Test turn on."""
-        self.device.send_key = mock.Mock()
-        self.device._mac = None
-        self.device.turn_on()
-        self.device.send_key.assert_called_once_with("KEY_POWERON")
-        self.device._wol.send_magic_packet = mock.Mock()
-        self.device._mac = "fake"
-        self.device.turn_on()
-        self.device._wol.send_magic_packet.assert_called_once_with("fake")
+@pytest.fixture(name="remote")
+def remote_fixture():
+    """Patch the samsungctl Remote."""
+    with patch("samsungctl.Remote") as remote_class, patch(
+        "homeassistant.components.samsungtv.media_player.socket"
+    ) as socket_class:
+        remote = mock.Mock()
+        remote_class.return_value = remote
+        socket = mock.Mock()
+        socket_class.return_value = socket
+        yield remote
+
+
+@pytest.fixture(name="wakeonlan")
+def wakeonlan_fixture():
+    """Patch the wakeonlan Remote."""
+    with MockDependency("wakeonlan") as wakeonlan:
+        yield wakeonlan
 
 
 @pytest.fixture
-def samsung_mock():
-    """Mock samsungctl."""
-    with patch.dict("sys.modules", {"samsungctl": MagicMock()}):
-        yield
+def mock_now():
+    """Fixture for dtutil.now."""
+    return dt_util.utcnow()
+
+
+async def setup_samsungtv(hass, config):
+    """Set up mock Samsung TV."""
+    await async_setup_component(hass, "media_player", config)
+    await hass.async_block_till_done()
 
 
-async def test_play_media(hass, samsung_mock):
+async def test_setup_with_mac(hass, remote):
+    """Test setup of platform."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert hass.states.get(ENTITY_ID)
+
+
+async def test_setup_duplicate(hass, remote, caplog):
+    """Test duplicate setup of platform."""
+    DUPLICATE = {DOMAIN: [MOCK_CONFIG[DOMAIN], MOCK_CONFIG[DOMAIN]]}
+    await setup_samsungtv(hass, DUPLICATE)
+    assert "Ignoring duplicate Samsung TV fake" in caplog.text
+
+
+async def test_setup_without_mac(hass, remote):
+    """Test setup of platform."""
+    await setup_samsungtv(hass, MOCK_CONFIG_NOMAC)
+    assert hass.states.get(ENTITY_ID_NOMAC)
+
+
+async def test_setup_discovery(hass, remote):
+    """Test setup of platform with discovery."""
+    hass.async_create_task(
+        async_load_platform(
+            hass, DOMAIN, SAMSUNGTV_DOMAIN, MOCK_CONFIG_DISCOVERY, {DOMAIN: {}}
+        )
+    )
+    await hass.async_block_till_done()
+    entity = hass.data[DOMAIN].get_entity(ENTITY_ID_DISCOVERY)
```

Comment:
We should get the state of the entity, not the entity itself. Use `hass.states.get`. We can assert the entity name via `state.name`.

To assert the entity unique_id, please follow this example:
https://github.com/home-assistant/home-assistant/blob/95295791bd4529d0e9435e03a46f4091e884fc04/tests/components/smartthings/test_fan.py#L64-L69